frontend/src/components/RegionMap.js

```diff
@@ -1,80 +1,92 @@
-import React, {useEffect, useRef} from 'react';
+import React, {useEffect, useRef, useState} from 'react';
 import maplibregl from 'maplibre-gl';
 import {useNavigation} from "./NavigationContext";
 import {fetchRegionGeometry} from "../api";
 
 const MapComponent = () => {
-    const mapContainer = useRef(null);
-    const map = useRef(null);
-    const { selectedRegion, selectedHierarchy } = useNavigation();
+  const mapContainer = useRef(null);
+  const map = useRef(null);
+  const { selectedRegion, selectedHierarchy } = useNavigation();
+  const [error, setError] = useState(null);
 
-    const fetchSelectedRegionGeometry = async () => {
-        if (selectedRegion.id !== null && selectedRegion.id !== 0) {
-            const response = await fetchRegionGeometry(selectedRegion.id, selectedHierarchy.hierarchyId);
-            if (response) {
-                return response.geometry;
-            } else {
-                return null;
-            }
+  const fetchSelectedRegionGeometry = async () => {
+    try {
+      if (selectedRegion.id !== null && selectedRegion.id !== 0) {
+        const response = await fetchRegionGeometry(selectedRegion.id, selectedHierarchy.hierarchyId);
+        if (response) {
+          return response.geometry;
+        } else {
+          throw new Error('No geometry data available for the selected region.');
```

Comment:
If the geometry does not exist for a region, it's not a critical error - we just do not render a map for it. It's checked further:
```js
const polygonData = await fetchSelectedRegionGeometry();

if (!polygonData || !polygonData.coordinates) {
  // Handle the case where there is no geometry data, perhaps set a default view?
  console.log('No geometry data available for the selected region.');
  return;
}
```
Later, we will implement a default view for this case. So, the absence of the geometry in the response should not set the error.